pkg/storage/tsdb/bucketindex/loader.go

```diff
@@ -215,7 +219,10 @@ func (l *Loader) updateCachedIndex(ctx context.Context, userID string) {
 	l.indexesMx.Unlock()
 
 	idx, err := ReadIndex(readCtx, l.bkt, userID, l.cfgProvider, l.logger)
-	if err != nil && !errors.Is(err, ErrIndexNotFound) && !errors.Is(err, bucket.ErrCustomerManagedKeyAccessDenied) {
+	if err != nil &&
+		!errors.Is(err, ErrIndexNotFound) &&
+		!errors.Is(err, bucket.ErrCustomerManagedKeyAccessDenied) &&
+		!errors.Is(err, context.Canceled) {
```

Comment:
What if the context is canceled due to `readIndexTimeout`? Do we want to cache that error in this case

Reply:
I think in this case is `context.DeadlineExceeded` error. But indeed we may dont wanna cache as well as it would just break something that was working before.

For now i left like this as this was never a problem

Reply:
I see